index.rst

```diff
@@ -240,6 +240,101 @@ fitSrcMatchAstrometry
+
+We lean towards merging PipelineTasks with the same dimensions that run back-to-back rather than keeping them distinct.
+This is a bit of a shift - many smaller PipelineTasks leads to more flexibility via just pipeline definition changes, which has been very useful in prototyping, but we believe we are exiting the prototyping phase and should instead prioritize the I/O optimization and pipeline-simplicity advantages of having fewer bigger PipelineTasks.
+We very much intend to continue to delegate all real algorithmic work to subtasks; it's just that each PipelineTask will tend towards having more of those.
+
+This is probably our last best chance to get our naming conventions for dataset types and task labels under control, so we're including that in our proposal.
+
+Conventions
+"""""""""""
+
+- Task labels are camelCase and start with a lowercase verb: "associateIsolatedStars" instead of "isolatedStarAssociation".
+- Dataset type names are snake_case nouns preceded (if necessary) by adjectives: e.g. "initial_visit_summary"
+- Use "source" instead of "src" or "sources".
+- Avoid "catalog" or "cat" in dataset type names; use "source", "stars", "object", or "matches" instead when appropriate.
+- Use some variant of "pvi" for any direct (non-difference) ``{visit, detector}`` image dataset with an image, mask, and variance plane (i.e. ``lsst.afw.image.Exposure`` or ``MaskedImage``).
+- Tables that are initially per-detector that will get concatenated into per-visit tables should get a "_detector" suffix so the final thing does not need a suffix (and when we revamp later steps of the pipeline, the same for "_patch" so there's no "_tract").
+- Task labels and dataset type names are for "slots", not specific tasks or connections - usually those are 1-1, but when they are not, the label and dataset type names should remain fixed when a different task is swapped in (our "solveAstrometry" task label slot could be satisfied by either jointcal or GBDES).
+  Whether the same is true of the outputs is an open question we'd like to discuss.
+- "initial" catalogs are not (ever) SDM-standardized, while final catalogs are always SDM-standardized before they are written out.
+  This may present a challenge for analysis tools that want to be able to run on either, but I'm hoping we can minimize analysis of the initial things, as we don't care how good they are in the end.
+- Do include a "final" prefix on dataset types that represent the best version of things that are nevertheless temporaries that will not be retained.
+  Do not add any prefix to dataset types that will be retained for public access.
+- Convert source and object catalogs to Parquet before ever persisting them; only use SourceCatalog/FITS to hold Footprints.
+  We will retain the ``slot_*`` column names and not the underlying names they point to - downstream code should only be referring to the slot columns anyway, and in many cases we won't run multiple algorithms that could satisfy the slot.
```

Comment:
Overall, I think we should re-visit the idea of slot altogether. I understand why it was introduced, but I think in practice measurements are not isometric to each other, and most downstream consumers actually care about the algorithm, such that they will always directly access that named column, or slots can never really be changed for how it will disrupt downstream consumers.

Reply:
I don't necessarily disagree but I'm considering that out of scope here.